File: java/src/com/novell/android/yastroid/ServerAddActivity.java
package com.novell.android.yastroid;

import static com.novell.android.yastroid.YastroidOpenHelper.*;

import java.net.URI;

import android.app.Activity;
import android.content.ContentValues;
import android.content.Intent;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class ServerAddActivity extends Activity {

	private SQLiteDatabase database;
	private Button addButton;
	private int groupId;

	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.serveradd);

		YastroidOpenHelper helper = new YastroidOpenHelper(this);
		database = helper.getWritableDatabase();

		Intent groupIntent = getIntent();
		Bundle b = groupIntent.getExtras();
		groupId = b.getInt("GROUP_ID");

		this.addButton = (Button)this.findViewById(R.id.button_add_server);
		this.addButton.setOnClickListener(new OnClickListener() {
			@Override
			public void onClick(View v) {
				if(addServer()) {
					finish();
				}
			}
		});
	}
	

	private boolean addServer() {
		boolean result = false;
		try {
			
			URI uri = new URI(((EditText)findViewById(R.id.edit_server_host)).getText().toString());
			String name = ((EditText)findViewById(R.id.edit_server_name)).getText().toString();
			String scheme = null;
			String host = uri.getHost();
			String port = ((EditText)findViewById(R.id.edit_server_port)).getText().toString();
			String user = ((EditText)findViewById(R.id.edit_server_user)).getText().toString();
			String pass = ((EditText)findViewById(R.id.edit_server_pass)).getText().toString();
			
			if (uri.getScheme() == null) {
				scheme = "http";
				host = uri.getSchemeSpecificPart();
			} else {
				scheme = uri.getScheme();
			}
			
			if (name.length() == 0 || host.length() == 0 || port.length() == 0 || user.length() == 0 || pass.length() == 0) {
				Toast.makeText(ServerAddActivity.this, "One or more fields are empty", Toast.LENGTH_SHORT).show();
				return false;
			}

			ContentValues values = new ContentValues();
			values.put(SERVERS_NAME, name);
			values.put(SERVERS_SCHEME, scheme);
			values.put(SERVERS_HOST, host);
			values.put(SERVERS_PORT, port);
			values.put(SERVERS_USER, user);
			values.put(SERVERS_PASS, pass);
			values.put(SERVERS_GROUP, groupId);
			
			database.insert(SERVERS_TABLE_NAME, "null", values);
			database.close();
			Log.i("ARRAY", "WebYaST server " + name + " has been added.");
			result = true;
		} catch (Exception e) {
			Log.e("BACKGROUND_PROC", e.getMessage());
		}
		Toast.makeText(ServerAddActivity.this, "Server Added", Toast.LENGTH_SHORT).show();
		return result;
	}
}

File: mono/YaSTroid/YaSTroid/ServerAddActivity.cs
using System;
using Android.App;
using Android.Content;
using Android.Database.Sqlite;
using Android.OS;
using Android.Util;
using Android.Widget;
using Java.Net;

namespace YaSTroid
{
	[Activity (Label = "ServerActivity")]
	public class ServerAddActivity : Activity
	{
		private SQLiteDatabase database;
		private Button addButton;
		private int groupId;

		protected override void OnCreate(Bundle savedInstanceState)
		{
			base.OnCreate(savedInstanceState);
			SetContentView(Resource.Layout.serveradd);

			YastroidOpenHelper helper = new YastroidOpenHelper(this);
			database = helper.WritableDatabase;

			Intent groupIntent = Intent;
			Bundle b = groupIntent.Extras;
			groupId = b.GetInt("GROUP_ID");

			this.addButton = FindViewById<Button>(Resource.Id.button_add_server);
			this.addButton.Click += (sender, e) => {
				if (addServer ()) {
					Finish ();
				}
			};				
		}
		

		private bool addServer()
		{
			bool result = false;
			try
			{	
				URI uri = new URI(FindViewById<EditText>(Resource.Id.edit_server_host).Text);
				string name = FindViewById<EditText>(Resource.Id.edit_server_name).Text;
				string scheme = null;
				string host = uri.Host;
				string port = FindViewById<EditText>(Resource.Id.edit_server_port).Text;
				string user = FindViewById<EditText>(Resource.Id.edit_server_user).Text;
				string pass = FindViewById<EditText>(Resource.Id.edit_server_pass).Text;
				
				if (uri.Scheme == null) {
					scheme = "http";
					host = uri.SchemeSpecificPart;
				} else {
					scheme = uri.Scheme;
				}
				
				if (name.Length == 0 || host.Length == 0 || port.Length == 0 || user.Length == 0 || pass.Length == 0) {
					Toast.MakeText(this, "One or more fields are empty", ToastLength.Short).Show();
					return false;
				}

				ContentValues values = new ContentValues();
				values.Put(YastroidOpenHelper.SERVERS_NAME, name);
				values.Put(YastroidOpenHelper.SERVERS_SCHEME, scheme);
				values.Put(YastroidOpenHelper.SERVERS_HOST, host);
				values.Put(YastroidOpenHelper.SERVERS_PORT, port);
				values.Put(YastroidOpenHelper.SERVERS_USER, user);
				values.Put(YastroidOpenHelper.SERVERS_PASS, pass);
				values.Put(YastroidOpenHelper.SERVERS_GROUP, groupId);
				
				database.Insert(YastroidOpenHelper.SERVERS_TABLE_NAME, "null", values);
				database.Close();
				Log.Info("ARRAY", "WebYaST server " + name + " has been added.");
				result = true;
			} catch (Exception e) {
				Log.Error("BACKGROUND_PROC", e.Message);
			}
			Toast.MakeText(this, "Server Added", ToastLength.Short).Show();
			return result;
		}
	}
}
